pipelines/pathway.go

```diff
@@ -0,0 +1,101 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package pipelines
+
+import (
+	"encoding/binary"
+	"hash/fnv"
+	"log"
+	"math/rand"
+	"time"
+)
+
+// Pathway represents a path points can take.
+// It is defined as nodes (services) linked together with edges.
```

Comment:
Please write more as if you would be writing a blog post about this structure, not "code documentation". By blog post I mean eligibility and educating for the user, not necessarily long 🙂